api/resource.rb

```diff
@@ -205,6 +205,16 @@ def required_properties
       all_user_properties.select(&:required)
     end
 
+    def all_nested_properties(props)
```

Comment:
It would be useful if this passed a `prefix` string through (such as `build_flatten_method` does), so that it returned `foo.bar` for nested fields instead of `bar`.